src/index.ts

```diff
@@ -99,6 +99,18 @@ class ExoplatformBot {
     }
   }
 
+  /**
+   * Make an API call to eXo Platform configured API, but throws if not logged in..
+   * @param args Same as `this.request`
+   * @returns Same as `this.request`
+   * @throws {Error} `this.login` must be called before `this.loggedInRequest`
+   */
+  private async loggedInRequest(...args: Parameters<ExoplatformBot['request']>): ReturnType<ExoplatformBot['request']> {
```

Comment:
That is cool, didn't knew about typescript private fields. I tried using Node v12 new private field syntax (`#field`), but it didnt work.
Maybe `exoHostname`, `exoPath`, `exoSecureProtocol`, `username`, `password` should all be private too.

Reply:
I was thinking it was intended so I don't touched them :smile: 
`exoHostname`, `exoPath` and `exoSecureProtocol` may stay `public` as their are user-submitted parameters, in the constructor moreover :thinking: 
But it's sure that `username` & `password` should be `private` as they require validation in `login`, and are sensitives informations :lock:

Reply:
I agree. Do these changes and I will merge it. Thanks!

Reply:
Done :smile: 
I just made use of some TypeScript magic
https://github.com/rigwild/exoplatform-api-wrapper/blob/4f8cf6b04e996c64b503d80952e9528cd797311b/src/index.ts#L16